### PR TITLE
EVA-3101 - Scripts to deprecate variants with allelesMatch false

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**

### CODE REVIEW

File: tasks/eva_3101/src/main/groovy/eva3101/eva3101_deprecate_allelesMatch_false_variants.groovy [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+def allelesMatchFalseVariantIterator = new RetryableBatchingCursor(where("allelesMatch").exists(true),
+        dbEnv.mongoTemplate, dbsnpSveClass).iterator()
+def dbEnvJobSteps = dbEnvStepBuilderFactory.get("stepsForEVA3101Deprecation").chunk(
+        new SimpleCompletionPolicy(1000)).reader(
+        new ItemStreamReader<SubmittedVariantEntity>() {
+            def currResultListIterator = null
+            @Override
+            SubmittedVariantEntity read() throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
+                // We need this somewhat convoluted read routine to
+                // unwind the "list of results" returned by the batching cursor
```
[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply tasks/eva_3101/src/main/groovy/eva3101/eva3101_deprecate_allelesMatch_false_variants.groovy:67]: Good point! See [this PR](https://github.com/EBIvariation/eva-common-groovyutils/pull/2)

File: tasks/eva_3101/src/main/groovy/eva3101/eva3101_deprecate_allelesMatch_false_variants.groovy [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+def dbEnv = createFromSpringContext(options.propertiesFile, Application.class)
+
+def dbEnvDeprecationWriter = new SubmittedVariantDeprecationWriter(assemblyToUse, dbEnv.mongoTemplate,
+        dbEnv.submittedVariantAccessioningService, dbEnv.clusteredVariantAccessioningService,
+        dbEnv.springApplicationContext.getBean("accessioningMonotonicInitSs", Long.class),
+        dbEnv.springApplicationContext.getBean("accessioningMonotonicInitRs", Long.class),
+        "EVA3101", "Variant deprecated due to alleles mismatch with the reference assembly.")
+def dbEnvProgressListener =
+        new DeprecationStepProgressListener(dbEnvDeprecationWriter, dbEnv.springApplicationContext.getBean(MetricCompute.class))
```
[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply tasks/eva_3101/src/main/groovy/eva3101/eva3101_deprecate_allelesMatch_false_variants.groovy:52]: Addressed..

File: tasks/eva_3101/src/main/groovy/eva3101/eva3101_prepare_devenv.groovy [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import uk.ac.ebi.eva.accession.deprecate.batch.listeners.DeprecationStepProgressListener
+import uk.ac.ebi.eva.groovy.commons.RetryableBatchingCursor
+import uk.ac.ebi.eva.metrics.metric.MetricCompute
+
+import java.time.LocalDateTime
+
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.*
+import static org.springframework.data.mongodb.core.query.Criteria.where
+
+// This script deprecates variants with allelesMatch attribute in dbsnpSVE collection and also the corresponding RS (if any)
```
[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply tasks/eva_3101/src/main/groovy/eva3101/eva3101_prepare_devenv.groovy]: Addressed..

File: tasks/eva_3101/src/main/groovy/eva3101/eva3101_deprecate_allelesMatch_false_variants.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (apriltuesday) [reply tasks/eva_3101/src/main/groovy/eva3101/eva3101_deprecate_allelesMatch_false_variants.groovy]: awesome, thanks!

**Review by apriltuesday — APPROVED** [review]

**Review by nitin-ebi — APPROVED** [review]

File: tasks/eva_3101/pom.xml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            <groupId>uk.ac.ebi.eva</groupId>
+            <artifactId>eva-common-groovyutils</artifactId>
+            <version>0.1-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>uk.ac.ebi.eva</groupId>
+            <artifactId>eva-accession-deprecate</artifactId>
+            <version>0.6.16-SNAPSHOT</version>
+            <exclusions>
+                <!-- Currently this is needed because the eva-accession-deprecate has a transitive dependency
```
Comment (nitin-ebi) [inline tasks/eva_3101/pom.xml:29]: Now that eva-metrics is using Spring Boot 2.1.0.RELEASE, this might not be required

**Pull request merged**